src/LayerTree/LayerTree.spec.jsx

```diff
@@ -312,6 +312,32 @@ describe('<LayerTree />', () => {
       expect(treeNode.props().checked).toBe(true);
     });
 
+    it('triggers tree rebuild on nodeTitleRenderer changes', () => {
+      const nodeTitleRenderer = function(layer) {
+        return (
+          <span className="span-1">
+            {layer.get('name')}
+          </span>
+        );
+      };
+
+      const props = {
+        layerGroup,
+        map,
+        nodeTitleRenderer
+      };
+      const wrapper = TestUtil.mountComponent(LayerTree, props);
+
+      const rebuildSpy = jest.spyOn(wrapper.instance(), 'rebuildTreeNodes');
```

Comment:
I would suggest to create the spy before adding the prop `nodeTitleRenderer` to the component. See [here](https://github.com/terrestris/react-geo/blob/master/src/Button/ToggleButton/ToggleButton.spec.jsx#L69) for an example.